### PR TITLE
LibWeb: Layout standalone SVG file obeys width/height attributes

### LINKED ISSUE #2668: Width/height attributes not respected for standalone SVG documents
### Summary

When opening in Ladybird an SVG file with specified width/height attributes  ("standalone SVG"), the svg drawing fills the whole agent viewport and does not respect the width/height. 

Chrome and Firefox respect the dimension attributes.

### Operating system

Linux

### Steps to reproduce

1. Open this in Ladybird:
https://github.com/user-attachments/assets/1b93deda-e7d3-42d9-a38a-5a45edcbb88c
2. ... then in Firefox
3. ... then Chromium

### Expected behavior

Ladybird should respect the width/height attributes of the SVG container.

### Actual behavior

The image fills the agent viewport.

### URL for a reduced test case

https://github.com/user-attachments/assets/1b93deda-e7d3-42d9-a38a-5a45edcbb88c

### HTML/SVG/etc. source for a reduced test case

```html
<svg viewBox="0 0 32 64" width="128" height="256" xmlns="http://www.w3.org/2000/svg">
    <title>white diamond on blue</title>
    <rect x="0" y="0" width="32" height="64" fill="blue"/>
    <g fill="white">
        <path d="M16 0 L32 32 L16 64 L0 32 Z" />
    </g>
</svg>
```

### Log output and (if possible) backtrace

```shell
# dump in headless browser

Viewport <#document> at (0,0) content-size 800x600 [BFC] children: not-inline
  SVGSVGBox <svg> at (0,0) content-size 800x600 [SVG] children: inline
    TextNode <#text>
    TextNode <#text>
    SVGGeometryBox <rect> at (250,0) content-size 300x600 children: not-inline
    TextNode <#text>
    SVGGraphicsBox <g> at (250,0) content-size 300x600 children: inline
      TextNode <#text>
      SVGGeometryBox <path> at (250,0) content-size 300x600 children: not-inline
      TextNode <#text>
    TextNode <#text>

ViewportPaintable (Viewport<#document>) [0,0 800x600]
  SVGSVGPaintable (SVGSVGBox<svg>) [0,0 800x600]
    SVGPathPaintable (SVGGeometryBox<rect>) [250,0 300x600]
    SVGGraphicsPaintable (SVGGraphicsBox<g>) [250,0 300x600]
      SVGPathPaintable (SVGGeometryBox<path>) [250,0 300x600]
```

### Screenshots or screen recordings

image
image
image

### Build flags or config settings

_No response_

### Contribute a patch?

- [X] I’ll contribute a patch for this myself.

### PR DESCRIPTION
**Pull request by manuel-za**
Fixes #2668 

- parity with Chromium (131.0.6778.85)
- very close to Firefox (129.0.2) -- difference not material

### CODE REVIEW

**Comment by circl-lastname** [comment]
Could you write "Fixes #2668"? That way Github automatically marks that issue resolved once merged

File: Libraries/LibWeb/Layout/SVGFormattingContext.cpp [hunk 1]
```diff
@@ -184,6 +184,21 @@ void SVGFormattingContext::run(AvailableSpace const& available_space)
...
+        // NOTE: If a height had not been provided by the svg element, it was set to the height of the container
+        //       (see BlockFormattingContext::layout_viewport)
+        if (svg_box_state.node().computed_values().width().is_length())
+            svg_box_state.set_content_width(svg_box_state.node().computed_values().width().length().absolute_length_to_px());
+        if (svg_box_state.node().computed_values().height().is_length())
+            svg_box_state.set_content_height(svg_box_state.node().computed_values().height().length().absolute_length_to_px());
+
+        // FIXME: find a more precise way to determine whether we are rendering a standalone SVG document
+        // Will break for image-SVG elements without a specified viewBox included in an XML document
+        // DOM::Document::is_xml_document basically means "it is not an html document" (https://dom.spec.whatwg.org/#xml-document)
```
Comment (AtkinsSJ) [inline Libraries/LibWeb/Layout/SVGFormattingContext.cpp:199]: Looks like for SVG documents we also set the content type, so you should be able to check `content_box().document().content_type() == "image/svg+xml"_string`.

Also, `<svg>` can appear inside another `<svg>`. If you want to only set the width and height for top-level SVGs, you'll need to add a check for that somehow.

**Review by AtkinsSJ — CHANGES_REQUESTED** [review]
Hi, welcome to Ladybird! :^)

A couple of more general comments:
- That first commit adds a test before the commit that makes the test pass. Please move it either into the second commit, or after it - all tests should pass when they're added.
- Commit titles should be imperatives (like a command). So, "Add unit tests..." instead of just "Unit tests...", and "Make standalone SVG files obey width..."

**Comment by manuel-za** [comment]
@AtkinsSJ : Thanks for the comments. Makes sense.
Rather than rewrite history + pollute the log, I created another PR addressing the comments: https://github.com/LadybirdBrowser/ladybird/pull/2806

I believe this new PR addresses your requested changes:
- unit tests come after the code change
- found a reliable way to detect running in the top formatting context

I would appreciate a review. Thanks! :]m

**Comment by manuel-za** [comment]
...closing and moving to #2806

**Comment by manuel-za** [comment]
@AtkinsSJ : Thanks for the comments. Makes sense.
Rather than rewrite history + pollute the log, I created another PR addressing the comments: https://github.com/LadybirdBrowser/ladybird/pull/2806

I believe this new PR addresses your requested changes:
- unit tests come after the code change
- found a reliable way to detect running in the top formatting context

I would appreciate a review. Thanks! :]m

**Pull request closed without merging**